release_process.md

```diff
@@ -56,3 +56,36 @@ want to increment the minor version, such as when there is a breaking change
 or a major new feature. This should be done manually in the PR that introduces
 the breaking change or new feature. Be sure to check that we haven't already
 incremented the minor version in the current release cycle.
+
+
+## Breaking Change policy
+
+We try to avoid breaking changes, but sometimes they are necessary. When there
+are breaking changes, we will increment the minor version. (This is valid 
+semantic versioning because we are still in `0.x` versions.)
+
+When a PR makes a breaking change, the PR author should mark the PR using the 
+conventional commit markers: either exclamation mark after the type
+(such as `feat!: change signature of func`) or have `BREAKING CHANGE` in the
+body of the PR. A CI job will add a `breaking-change` label to the PR, which is
+what will ultimately be used to CI to determine if the minor version should be
+incremented.
+
+A CI job will validate that if a `breaking-change` label is added, the minor
+version is incremented in the `Cargo.toml` and `pyproject.toml` files. The only
+exception is if it has already been incremented since the last stable release.
+
+**It is the responsibility of the PR author to increment the minor version when
+appropriate.**
+
+Some things that are considered breaking changes:
+
+* Upgrading a dependency pin that is in the Rust API. In particular, upgrading
+  `DataFusion` and `Arrow` are breaking changes. Changing dependencies that are
+  not exposed in our public API are not considered breaking changes.
```

Comment:
Technically this is only true if we are upgrading from one major version to another.  However, given that has been the case for most releases of arrow and datafusion, I'm not sure it needs to be called out.

Reply:
Depends on our definition of breaking change. For example if we change the pin from `40.0.0` to `40.0.1`, and a user has pinned `=40.0.0`, we will have broken their build. But not really made a meaningful break change behavior wise I think.